contracts/liquidators/SolidlyLpTokenLiquidator.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.0;
+
+import "openzeppelin-contracts-upgradeable/contracts/token/ERC20/IERC20Upgradeable.sol";
+import "openzeppelin-contracts-upgradeable/contracts/token/ERC20/utils/SafeERC20Upgradeable.sol";
+
+import "../external/solidly/IRouter.sol";
+import "../external/solidly/IPair.sol";
+
+import "./IRedemptionStrategy.sol";
+
+/**
+ * @title SolidlyLpTokenLiquidator
+ * @notice Exchanges seized Solidly LP token collateral for underlying tokens for use as a step in a liquidation.
+ */
+contract SolidlyLpTokenLiquidator is IRedemptionStrategy {
+  using SafeERC20Upgradeable for IERC20Upgradeable;
+
+  /**
+   * @dev Internal function to approve unlimited tokens of `erc20Contract` to `to`.
+   */
+  function safeApprove(
```

Comment:
this safeApprove fn is pointless to use when the token transfer is in a single tx